src/services/policies/mod.rs

```diff
@@ -14,12 +15,21 @@ pub trait PolicyStore: Send + Sync {
     async fn policy_set(&self) -> PolicySet;
     async fn get_policies(&self) -> Vec<Policy>;
     async fn get_policy(&self, id: &str) -> Result<Policy, Box<dyn Error>>;
-    async fn create_policy(&self, policy: &Policy) -> Result<Policy, Box<dyn Error>>;
-    async fn update_policies(&self, policies: Vec<Policy>) -> Result<Vec<Policy>, Box<dyn Error>>;
+    async fn create_policy(
+        &self,
+        policy: &Policy,
+        schema: Option<Schema>
```

Comment:
instead of having to pass the schema store to every function call, you can also make this a dependency in the PolicyStore init - for example initializing the MemoryPolicyStore with the MemorySchemaStore ( should accept the trait, not the impl )

Reply:
I'm open to the idea but I think there is a potential issue? In the rest api endpoints we are passing schemas from the memory store created by rocket, which can be interacted with through its own endpoints. Doesn't it make more sense for them to exist separately, rather than the schema store being a member of the policy store? I also just like being able to pass an arbitrary schema from the outside

Reply:
agree with you, anyway this was just a semantic :)